crates/sqlexec/src/metastore/catalog.rs

```diff
@@ -436,24 +437,34 @@ impl NamespacedCatalogEntry<'_> {
 #[derive(Debug, Default)]
 pub struct TempObjects {
     /// In-memory (temporary) tables.
-    current_session_tables: HashMap<String, Arc<MemTable>>,
+    current_session_tables: RwLock<HashMap<String, Arc<MemTable>>>,
```

Comment:
why not use `DashMap`? it's designed to be a replacement for this. 

> DashMap tries to be very simple to use and to be a direct replacement for RwLock<HashMap<K, V, S>>. To accomplish this, all methods take &self instead of modifying methods taking &mut self. This allows you to put a DashMap in an Arc<T> and share it between threads while being able to modify it.

Reply:
DashMap would actually incur more overhead here, specifically when returning the names of all tables, as it would need to acquire multiple locks.

I think the rwlock here is fine since these are scoped to a single session, and so concurrent access is unlikely (so much so that I would actually prefer a plain mutex here, but that's minor).